stellargraph/core/graph.py

```diff
@@ -832,6 +833,17 @@ def _unique_type_triples(self, *, return_counts, selector=slice(None)):
 
         return zip(*unique_ets)
 
+    def _edge_metrics_by_type_triple(self, *, selector=slice(None), metrics):
+        src_ty, rel_ty, tgt_ty = self._edge_type_triples(selector)
+        df = pd.DataFrame(
+            self._edges.weights, index=[src_ty, rel_ty, tgt_ty], columns=["weight"]
+        )
+
+        if len(df):
```

Comment:
group-by doesn't work with an empty dataframe? It's unfortunate that it doesn't just give an empty dataframe in return?

Reply:
hmm I think it's the `agg` that fails due to not having numeric columns to aggregate, and actually it seems to work if I explicitly cast the `weight` column to be numeric
```python
df["weights"] = pd.to_numeric(df["weights"])
```
so I'll switch to that and just return a dataframe

Reply:
Ah, hm. With the new form using `dict`, it seems like we may not need this work around as aggregation works directly:

```python
pd.DataFrame({"a": [], "b": []}).groupby("a").agg("mean")
# Empty DataFrame
# Columns: [b]
# Index: []
```

It seems like the exception you mention only occurs when using `columns=[...]` to set the names like the old form:

```python
pd.DataFrame([], columns=["a", "b"]).groupby("a").agg("mean")
# pandas.core.base.DataError: No numeric types to aggregate
```

Reply:
Seems like `graph._edges.weights` get a dtype of `object` if the graph was created with no edges.. looking into where to do the casting

Reply:
This might be the same problem as https://github.com/stellargraph/stellargraph/pull/1220#discussion_r409221636

Reply:
I've added that one line change here since #1220 won't be landing for the RC

Reply:
actually this is no longer needed after #1284 🎉